### PR TITLE
Handle narrow to unsubscribed stream.

### PR DESCRIPTION
**Pull request by vishwesh3**
part of #564

### CODE REVIEW

File: src/message/NotSubscribed.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  render() {
+    const { showSubscribeButton } = this.props;
+
+    return (
+      <View style={styles.container}>
+        <Label
+          style={styles.text}
+          text={"You aren't subscribed to this stream."}
```
Comment (borisyankov) [inline src/message/NotSubscribed.js]: Just rename to 'are not' and do not add the 'aren' to the dictionary.

File: src/utils/__tests__/narrow-test.js [hunk 2]
```diff
@@ -108,6 +109,17 @@ describe('isPrivateOrGroupNarrow', () => {
   });
 });
 
+describe('isStreamOrTopicNarrow', () => {
```
Comment (borisyankov) [inline src/utils/__tests__/narrow-test.js]: I like tests ;)

File: src/title/TitleStream.js [hunk 3]
```diff
@@ -18,17 +18,20 @@ export default class TitleStream extends React.PureComponent {
...
   render() {
-    const { narrow, subscriptions, color } = this.props;
-    const stream = subscriptions.find(x => x.name === narrow[0].operand) || {
-      name: '',
-      invite_only: false,
-      in_home_view: false
-    };
+    const { narrow, subscriptions, streams, color } = this.props;
+    const stream = subscriptions.find(x => x.name === narrow[0].operand) ||
+      { ...streams.find(x => x.name === narrow[0].operand), in_home_view: true } ||
```
Comment (borisyankov) [inline src/title/TitleStream.js]: Do you need to search both subscriptions and streams?
Streams contain all items already.

I am also suspicious of the use of `in_home_view` in this way...

File: src/message/NotSubscribed.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  auth: Auth,
+  narrow: Narrow,
+  showSubscribeButton: boolean,
+};
+
+class NotSubscribed extends React.Component {
+
+  props: Props;
+
+  subscribeStream = () => {
```
Comment (borisyankov) [inline src/message/NotSubscribed.js]: subscribeStream => subscribeToStream

File: src/message/NotSubscribed.js (hunk 1 above)
Comment (vishwesh3) [inline src/message/NotSubscribed.js]: Done 👍

File: src/title/TitleStream.js (hunk 3 above)
Comment (vishwesh3) [inline src/title/TitleStream.js]: Yes streams contains all items, but it does not contain `in_home_view` which is used to determine stream icon [here](https://github.com/zulip/zulip-mobile/blob/master/src/title/TitleStream.js#L43)

File: src/message/NotSubscribed.js (hunk 4 above)
Comment (vishwesh3) [inline src/message/NotSubscribed.js]: Done 👍

**Comment by vishwesh3** [comment]
@borisyankov rebased 👍

File: src/title/TitleStream.js (hunk 3 above)
Comment (borisyankov) [inline src/title/TitleStream.js]: Maybe we should merge the data for both, like 'isSubscribed' or something to determine 'subscriptions' as a subset of streams.

Comment (borisyankov) [inline src/title/TitleStream.js]: Consider that, but I'll merge this PR, as it is now good.

**Pull request merged**